SignalServiceKit/src/Messages/OWSReadReceiptManager.m

```diff
@@ -343,10 +310,10 @@ - (void)messageWasReadLocally:(TSIncomingMessage *)message
             NSString *messageAuthorId = message.messageAuthorId;
             OWSAssert(messageAuthorId.length > 0);
 
-            OWSReadReceipt *newReadReceipt =
-                [[OWSReadReceipt alloc] initWithSenderId:messageAuthorId timestamp:message.timestamp];
+            OWSLinkedDeviceReadReceipt *newReadReceipt =
+                [[OWSLinkedDeviceReadReceipt alloc] initWithSenderId:messageAuthorId timestamp:message.timestamp];
 
-            OWSReadReceipt *_Nullable oldReadReceipt = self.toLinkedDevicesReadReceiptMap[threadUniqueId];
+            OWSLinkedDeviceReadReceipt *_Nullable oldReadReceipt = self.toLinkedDevicesReadReceiptMap[threadUniqueId];
             if (oldReadReceipt && oldReadReceipt.timestamp > newReadReceipt.timestamp) {
                 // If there's an existing read receipt for the same thread with
```

Comment:
As I understand it, we want to change this behavior, to send a timestamp for every individual message read. Right?

Reply:
* "linked device" read receipts do "implication" (e.g. affect older message as well).  
* "sender/recipient" read receipts do not.  

These are the latter, so this is correct.  I've clarified the comment.

Reply:
Oh right, sorry.